### ISSUE TITLE
Ember addon stubbing issues

### ISSUE THREAD

**Issue by machty** [isleofcode/corber#403]
Collecting related issues in this GH issue since I think they might all gravitate around a similar solution:

This repo [injects a fake POJO Addon](https://github.com/isleofcode/corber/blob/master/lib/frameworks/ember/tasks/serve.js#L37) into Ember's build pipline during `corber s` to mix into platform-specific cordova.js/cordova_plugins.js into the app's JS payload. I've run into a few problems with this approach:

1. ember-intl searches through all addons for translation files, but barfs at [this line](https://github.com/ember-intl/ember-intl/blob/8732c5d6a1dcbfca23464fd17aade609b45f0dd1/index.js#L233) because it's expecting more than just the `treeFor` property on the "addon". I ended up having to fork corber and add a few more stubbed properties just to get it to compile:

```js
...

        return pluginsTree;
      },
      pkg: {
        "ember-addon": {}
      },
      root: "nonexistent",
      name: "stubbed-corber-live-reload-injected-plugin",
      addons: []
    });
```

2. Minor point but something I got stuck on for longer than I care to admit: me and my team missed the memo about the [defaults pojo that's passed to ember-cli-build](https://github.com/ember-cli/ember-cli/commit/71f9bd22cf4f321f8e81bbf93db5fedfbf321288) that is expected to be passed to the EmberApp that's returned from the fn, and corber is the first addon we've run into that broke due to us omitting that. Specifically, the project that corber clones is ultimately passed in as part of the `defaults`, so if you don't use defaults, the `corber s` won't ever call the `treeFor` hook.

3. `treeFor` might be called multiple times, passing in a different "treeName" arg that it wants; in my app, this fn gets called 3 times w the following respective args: "app", "templates", "vendor". Because the cordova/cordova_plugins code was being included in our codebase in 3 different places, it was causing errors where "window.cordova is already defined". I added (to my fork) a conditional check to make return null unless `treeName === 'app'`. I also just tried defining `treeForVendor` at one point but it wasn't getting called (there might be some internal ember-cli logic that checks/expects `treeFor` to be defined.

### Conclusion

Not sure, but likely need to find something less stubby/hacky that involves real EmberAddon objects and follows the happy path a bit better.

**Comment by ddoria921** [comment]
@machty I ran into the same `ember-intl` myself. @alexblom any chance we can get his fix merged in if there's a PR?

**Comment by alexblom** [comment]
Fixed in 1.3.0
